Riot/Modules/MatrixKit/Utils/EventFormatter/MXKEventFormatter.m

```diff
@@ -1762,7 +1762,32 @@ - (NSAttributedString*)renderHTMLString:(NSString*)htmlString forEvent:(MXEvent*
     // Note: `isEditMode` fixes an issue where editing a reply would display an "In reply to" span instead of a mention.
     if (!isEditMode && (event.isReplyEvent || (!RiotSettings.shared.enableThreads && event.isInThread)))
     {
-        html = [self renderReplyTo:html withRoomState:roomState];
+        MXEvent *repliedEvent = [self->mxSession.store eventWithEventId:event.relatesTo.inReplyTo.eventId inRoom:roomState.roomId];
+        if (repliedEvent)
+        {
+            NSString *repliedEventContent;
+            NSString *eventContent;
+            MXJSONModelSetString(repliedEventContent, repliedEvent.content[kMXMessageBodyKey]);
+
+            if (event.content[kMXMessageContentKeyNewContent])
+            {
+                MXJSONModelSetString(eventContent, event.content[kMXMessageContentKeyNewContent][@"formatted_body"]);
+                if (!eventContent)
+                {
+                    MXJSONModelSetString(eventContent, event.content[kMXMessageContentKeyNewContent][kMXMessageBodyKey]);
+                }
+                html = [NSString stringWithFormat:@"<mx-reply><blockquote><a href=\"%@\">In reply to</a> <a href=\"%@\">%@</a><br>%@</blockquote></mx-reply>%@",
+                        [MXTools permalinkToEvent:repliedEvent.eventId inRoom:repliedEvent.roomId],
+                        [MXTools permalinkToUserWithUserId:repliedEvent.sender],
+                        repliedEvent.sender,
+                        repliedEventContent,
+                        eventContent];
+            }
```

Comment:
if there is no `new_content` for this repliedEvent, you should execute:
`html = [self renderReplyTo:html withRoomState:roomState];`

Reply:
Thanks.

Yes indeed, this is actually the very annoying part because we're re-writing the html but we don't have a specific html representation for a deleted event, which means we either need a way to bring the info to the post-attributedstring generation or rebrand a lot more things in the formatter.